### PR TITLE
docs: update pull request template and specify squash merge

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- updates this pull request template with a test plan

### Test Plan:

- [ ] Wrote [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [x] CI tests / new tests are not applicable
- [ ] Manually tested my changes, but I want to keep the details secret
- [ ] Manually tested my changes, and here are the details:
  - Create an [alpha publish](https://github.com/chanzuckerberg/edu-design-system/blob/main/docs/PUBLISHING.md#alpha-release) and try out in `edu-stack` or `traject` as a sanity check if changes affect build or deploy, or are breaking.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+
+<!--
+  How did you validate that your changes were implemented correctly?
+-->
+
+- [ ] Wrote [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
+- [ ] CI tests / new tests are not applicable
+- [ ] Manually tested my changes, but I want to keep the details secret
+- [ ] Manually tested my changes, and here are the details:
+  - Create an [alpha publish](https://github.com/chanzuckerberg/edu-design-system/blob/main/docs/PUBLISHING.md#alpha-release) and try out in `edu-stack` or `traject` as a sanity check if changes affect build or deploy, or are breaking.
```
Comment (jeremiah-clothier) [inline .github/pull_request_template.md:13]: Is it mainly intuition for when this level of testing is needed or are there common things that have caused problems in the past and would be helpful to capture (if not already captured)?

Comment (jinlee93) [reply .github/pull_request_template.md:13]: I think this is mostly useful for breaking changes, such as token changes, widely used component updates, hooks changes, major deps upgrades, which I can add to this bullet

**Review by jeremiah-clothier — APPROVED** [review]

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -22,4 +22,4 @@ In order to contribute to EDS, please follow this workflow:
 2. Do feature work in accordance to the library's [code guidelines](./CODE_GUIDELINES.md). EDS uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), so each commit should follow commit guidelines reflecting the nature of the work.
 3. When work is ready for review, issue a pull request into `next` following the pull request guidelines. Tag `chanzuckerberg/edu-design-system` as reviewers for every PR.
 4. The library owners and trusted contributors will review the work and suggest revisions or approve the pull request
-5. Upon approval, feature gets merged into `next` to be slated for the next library release
+5. Upon approval, feature gets **Squash and merged** into `next` to be slated for the next library release
```
Comment (jeremiah-clothier) [inline docs/CONTRIBUTING.md:25]: Should we update the repository settings to not allow merge commits? https://github.com/chanzuckerberg/edu-design-system/settings

image

Comment (jinlee93) [reply docs/CONTRIBUTING.md:25]: Merge commits are still used for merging [release branches](https://github.com/chanzuckerberg/edu-design-system/blob/main/docs/PUBLISHING.md?plain=1#L57) into `main`

**Pull request merged**